### PR TITLE
Remove Mac OS 10.10, 10.11 and Debian 7

### PR DESCRIPTION
**Pull request by ekinanp**

### CODE REVIEW

**Comment by ekinanp** [comment]
Updated the PR to re-order the commits so that the removal from build-defaults comes before the removal of the platform.rb file.

**Comment by branan** [comment]
Is the related CJC PR up for this so we can land them together for tonight's runs?

**Comment by ekinanp** [comment]
@branan Yup, that was merged in earlier -- https://github.com/puppetlabs/ci-job-configs/pull/4377

File: configs/components/facter.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@
   case platform.name
   when /el-(6|7)/
     pkg.build_requires 'java-1.8.0-openjdk-devel'
-  when /(debian-(7|8)|ubuntu-14)/
+  when /(debian-(8)|ubuntu-14)/
```
Comment (branan) [inline configs/components/facter.rb]: Don't need the paren around `8` anymore

File: configs/components/facter.rb [hunk 2]
```diff
@@ -58,7 +58,7 @@
   case platform.name
   when /el-(6|7)/
     pkg.build_requires 'java-1.8.0-openjdk-devel'
-  when /(debian-(7|8)|ubuntu-14)/
+  when /(debian-8|ubuntu-14)/
```
Comment (ekinanp) [inline configs/components/facter.rb]: Removed the paren.

**Comment by ekinanp** [comment]
I'm gonna go ahead and add some more platform removals to this PR.

**Comment by ekinanp** [comment]
jenkins please test this on osx1012-64a

**Comment by ekinanp** [comment]
^ PR test is to make sure that small packaging change doesn't break anything

**Pull request merged**